client/app/shared/modals/data-upload/data-upload.component.ts

```diff
@@ -1,35 +1,48 @@
-import { Component } from '@angular/core'
+import { Component, OnInit, Input } from '@angular/core'
 import { UploadEvent, UploadFile, FileSystemFileEntry } from 'ngx-file-drop'
+import { forkJoin } from 'rxjs'
 
 import { FileType } from '../../enums/file-type'
 import { Team } from '../../models/team'
 import { ParserService } from '../../services/parser.service'
 import { TournamentDataService } from '../../services/tournament-data.service'
 import { TeamsService } from '../../services/teams.service'
+import { MatchesService } from '../../services/matches.service'
 import { Notifications } from '../../services/notifications.service'
 import { LoggerService } from '../../services/logger.service'
 
+declare var jQuery: any
```

Comment:
jQuery is used but not with a require call also it's not part of the package.json.

Reply:
You're right. It's part of the Semantic pacakge. Thereafore, it only needs to be declared.

Reply:
It's a very bad practice, to use it that way, we should declare those as peer / optional dependencies. and require them normally. Also you might not need jQuery

Reply:
I checked... I need it. I tried for a month without it before finally giving up and requiring the whole thing in the UI package. Peer dependency might be the correct call here, but you don't really need it in production, since you use the already build version of the UI package, which includes jQuery in it.